lib/cli/ui/prompt/interactive_options.rb

```diff
@@ -157,12 +165,43 @@ def raw_tty!
           end
         end
 
+        def presented_options(recalculate: false)
```

Comment:
this is really hard to follow.
I feel like it would do well to have some comments throughout explaining the code as it's quite logic dense and sort of magic with the `[-2].last` etc